compression/compression.go

```diff
@@ -32,3 +37,40 @@ func Compress(archivePath string, includePaths []string, logger log.Logger, envR
 
 	return nil
 }
+
+func AreAllPathsEmpty(includePaths []string) bool {
+	allEmpty := true
+
+	for _, path := range includePaths {
+		// Check if file exists at path
+		fileInfo, err := os.Stat(path)
+		if errors.Is(err, fs.ErrNotExist) {
+			// File doesn't exist
+			continue
+		}
+
+		// Check if it's a directory
+		if !fileInfo.IsDir() {
+			// Is a file and it exists
+			allEmpty = false
+			break
+		}
+
+		file, err := os.Open(path)
+		if err != nil {
+			continue
+		}
+		_, err = file.Readdirnames(1) // query only 1 child
+		if errors.Is(err, io.EOF) {
+			// Dir is empty
+			continue
+		}
+		if err == nil {
+			// Dir has files or dirs
+			allEmpty = false
```

Comment:
just to double check: this means that tar won't fail if there are only empty dirs inside the paths, right?

Reply:
the tests answer this, ignore :)